tailmac/Swift/Common/VnetHostConfigHelper.swift

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+import Foundation
+import Virtualization
+
+struct VnetHostConfigHelper {
+    let config: Config
+
+    func computeCPUCount() -> Int {
+        let totalAvailableCPUs = ProcessInfo.processInfo.processorCount
+
+        var virtualCPUCount = totalAvailableCPUs <= 1 ? 1 : totalAvailableCPUs - 1
+        virtualCPUCount = max(virtualCPUCount, VZVirtualMachineConfiguration.minimumAllowedCPUCount)
+        virtualCPUCount = min(virtualCPUCount, VZVirtualMachineConfiguration.maximumAllowedCPUCount)
+
+        return virtualCPUCount
+    }
+
+    func computeMemorySize() -> UInt64 {
+        // Set the amount of system memory to 4 GB; this is a baseline value
+        // that you can change depending on your use case.
+        var memorySize = config.memorySize
+        memorySize = max(memorySize, VZVirtualMachineConfiguration.minimumAllowedMemorySize)
+        memorySize = min(memorySize, VZVirtualMachineConfiguration.maximumAllowedMemorySize)
+
+        return memorySize
+    }
+
+    func createBootLoader() -> VZMacOSBootLoader {
+        return VZMacOSBootLoader()
+    }
+
+    func createGraphicsDeviceConfiguration() -> VZMacGraphicsDeviceConfiguration {
+        let graphicsConfiguration = VZMacGraphicsDeviceConfiguration()
+        graphicsConfiguration.displays = [
+            // The system arbitrarily chooses the resolution of the display to be 1920 x 1200.
+            VZMacGraphicsDisplayConfiguration(widthInPixels: 1920, heightInPixels: 1200, pixelsPerInch: 80)
+        ]
+
+        return graphicsConfiguration
+    }
+
+    func createBlockDeviceConfiguration() -> VZVirtioBlockDeviceConfiguration {
+        guard let diskImageAttachment = try? VZDiskImageStorageDeviceAttachment(url: config.diskImageURL, readOnly: false) else {
```

Comment:
Let's replace this with a `try-catch` so we can log the error instead of `try?`.